File: Leetcode/sbgo/Leetcode1004-最大连续1的个数III.go
package sbgo

func longestOnes(A []int, K int) int {
	start, ans, n := 0, 0, len(A)
	zeros := 0
	for end := 0; end < n; end++ {
		if A[end] == 0 {
			zeros++
		}
		for zeros > K {
			if A[start] == 0 {
				zeros--
			}
			start++
		}
		ans = max(ans, end-start+1)
	}
	return ans
}

File: Leetcode/PY/Leetcode1004-最大连续1的个数III.py
from typing import List
class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        start,zeros,ans = 0,0,0
        n = len(A)
        for end in range(n):
            if A[end] == 0:
                zeros += 1
            while zeros > K:
                if A[start] == 0:
                    zeros -= 1
                start += 1
            ans = max(ans,end - start + 1)
        return ans